sdk/test-utils/recorder/src/baseRecorder.ts

```diff
@@ -6,30 +6,33 @@ import nise from "nise";
 import {
   isBrowser,
   blobToString,
-  escapeRegExp,
   env,
   TestInfo,
   parseUrl,
   isPlaybackMode,
   isRecordMode,
-  findRecordingsFolderPath
+  findRecordingsFolderPath,
+  applyReplacementDictionary,
+  applyReplacementFunctions,
+  ReplacementFunctions,
+  ReplacementDictionary
 } from "./utils";
 import { customConsoleLog } from "./customConsoleLog";
 
 let nock: any;
 
-let replaceableVariables: { [x: string]: string } = {};
-export function setReplaceableVariables(a: { [x: string]: string }): void {
-  replaceableVariables = a;
+let replaceableVariables: ReplacementDictionary = {};
```

Comment:
any thoughts on making this a `Map<>` instead of a plain object?

Reply:
Maps might be a bit alienating to the JavaScript consumers, I would think. I can do the change if you're convinced it will be better. Just asking for a final thought.

Reply:
Eh, it's been around since IE11 and this is an internal package, right? I'll let @bterlson weigh in though.

Reply:
So, DefinitelyTyped uses NodeJS.ProcessEnv. I'll use that one. https://github.com/DefinitelyTyped/DefinitelyTyped/blob/a277fa20b586433979d6b5dd61b6fac42cc724f1/types/node/v10/child_process.d.ts#L343

Reply:
Maps are not alienating I think. Perhaps the opposite these days, since it provides a simpler semantics for sets than using objects for same.

Just my opinion though. I'd prefer a map I think.

Reply:
These values come from something that's being used by several of our clients at the moment. I need to add a compatibility layer to transform those plain objects into maps. That won't be a problem though, but just so that you're aware.

Reply:
The editor is telling me that I need to call TypeScript's compiler with `--downlevelIteration` to iterate over maps. Should I worry about that or not?

Reply:
Done!